packages/flutter/test/material/segmented_button_test.dart

```diff
@@ -1122,6 +1122,44 @@ void main() {
         ),
     );
   }, skip: kIsWeb && !isSkiaWeb); // https://github.com/flutter/flutter/issues/99933
+
+  testWidgets('SegmentedButton vertical aligned children', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          body: Center(
+            child: SegmentedButton<int>(
+              segments: const <ButtonSegment<int>>[
+                ButtonSegment<int>(
+                  value: 0,
+                  label: Text('Option 1'),
+                ),
+                ButtonSegment<int>(
+                  value: 1,
+                  label: Text('Option 1'),
+                ),
+                ButtonSegment<int>(
+                  value: 2,
+                  label: Text('Option 3'),
+                ),
+                ButtonSegment<int>(
+                  value: 3,
+                  label: Text('Option 4'),
+                ),
+              ],
+              onSelectionChanged: (Set<int> selected) {},
+              selected: const <int>{0},
+              direction: Axis.vertical,
+            ),
+          ),
+        ),
+      ),
+    );
+
+    final Finder button = find.byType(SegmentedButton<int>);
+    expect(tester.getSize(button).height,
```

Comment:
I think we might also want to check whether each segmented is aligned vertically with the same left and right, and the bottom of the previous one should be the top of the next one. I think `tester.getRect(find.byType(SegmentedButton<int>))` can be used to get the whole segmented button's size and position and tester.getRect(find.widgetWithText(TextButton, 'Option x')) can be used to check whether each segment inside has correct location and size:)

Reply:
Nice tips, am going to work on it.

Reply:
@QuncCccccc there is something weird going on when i used `tester.getRect(find.widgetWithText(TextButton, 'Option 0'));` to check the first element ,  the test finder cannot detect  ButtonSegment if its value is zero

Reply:
I find out that `TextButton` is removed underneath from (selected) segment button.

Reply:
I found a solution , by set `selected` to unknown value in the test, to avoid `TextButton` lost from the elements tree, every thing fine now 👍 .